src/base/process_util.c

```diff
@@ -0,0 +1,357 @@
+
+      SetHandleInformation (hStdIn, HANDLE_FLAG_INHERIT, HANDLE_FLAG_INHERIT);
+      start_info.dwFlags = STARTF_USESTDHANDLES;
+      start_info.hStdInput = hStdIn;
+      inherit_flag = TRUE;
+    }
+  if (stdout_file)
+    {
+      hStdOut =
+	CreateFile (stdout_file, GENERIC_WRITE, FILE_SHARE_READ, NULL, CREATE_ALWAYS, FILE_ATTRIBUTE_NORMAL, NULL);
+      if (hStdOut == INVALID_HANDLE_VALUE)
+	{
+	  assert (false);
+	  return 1;
+	}
+      SetHandleInformation (hStdOut, HANDLE_FLAG_INHERIT, HANDLE_FLAG_INHERIT);
+      start_info.dwFlags = STARTF_USESTDHANDLES;
+      start_info.hStdOutput = hStdOut;
+      inherit_flag = TRUE;
+    }
+  if (stderr_file)
+    {
+      hStdErr =
+	CreateFile (stderr_file, GENERIC_WRITE, FILE_SHARE_READ | FILE_SHARE_WRITE | FILE_SHARE_DELETE, NULL,
+		    CREATE_ALWAYS, FILE_ATTRIBUTE_NORMAL, NULL);
+      if (hStdErr == INVALID_HANDLE_VALUE)
+	{
+	  assert (false);
+	  return 1;
+	}
+
+      SetHandleInformation (hStdErr, HANDLE_FLAG_INHERIT, HANDLE_FLAG_INHERIT);
+      start_info.dwFlags = STARTF_USESTDHANDLES;
+      start_info.hStdError = hStdErr;
+      inherit_flag = TRUE;
+    }
+
+  res =
+    CreateProcess (argv[0], cmd_arg_ptr, NULL, NULL, inherit_flag, CREATE_NO_WINDOW, NULL, NULL, &start_info,
+		   &proc_info);
+  free (cmd_arg_ptr);
+
+  if (res == FALSE)
+    {
+      return 1;
+    }
+
+  if (hStdIn != INVALID_HANDLE_VALUE)
+    {
+      rc = CloseHandle (hStdIn);
+      if (rc == FALSE)
+	{
+	  assert (false);
+	  return 1;
+	}
+    }
+  if (hStdOut != INVALID_HANDLE_VALUE)
+    {
+      rc = CloseHandle (hStdOut);
+      if (rc == FALSE)
+	{
+	  assert (false);
+	  return 1;
+	}
+    }
+  if (hStdErr != INVALID_HANDLE_VALUE)
+    {
+      rc = CloseHandle (hStdErr);
+      if (rc == FALSE)
+	{
+	  assert (false);
+	  return 1;
+	}
+    }
+
+  new_pid = proc_info.dwProcessId;
+
+  if (wait_flag)
+    {
+      DWORD status = 0;
+
+      status = WaitForSingleObject (proc_info.hProcess, INFINITE);
+      if (status == WAIT_FAILED)
+	{
+	  assert (false);
+	  return 1;
+	}
+      rc = GetExitCodeProcess (proc_info.hProcess, &status);
+      if (rc == FALSE)
+	{
+	  assert (false);
+	  return 1;
+	}
+      if (exit_status != NULL)
+	{
+	  *exit_status = status;
+	}
+      rc = CloseHandle (proc_info.hProcess);
+      if (rc == FALSE)
+	{
+	  assert (false);
+	  return 1;
+	}
+      rc = CloseHandle (proc_info.hThread);
+      if (rc == FALSE)
+	{
+	  assert (false);
+	  return 1;
+	}
+      return 0;
+    }
+  else
+    {
+      rc = CloseHandle (proc_info.hProcess);
+      if (rc == FALSE)
+	{
+	  assert (false);
+	  return 1;
+	}
+      rc = CloseHandle (proc_info.hThread);
+      if (rc == FALSE)
+	{
+	  assert (false);
+	  return 1;
+	}
+      return new_pid;
+    }
+}
+#else
+  int pid, rc;
+
+  if (exit_status != NULL)
+    {
+      *exit_status = 0;
+    }
+
+  if (wait_flag)
+    {
+      if (signal (SIGCHLD, SIG_DFL) == SIG_ERR)
+	{
+	  assert (false);
+	  return 1;
+	}
+    }
+  else
```

Comment:
I don't know there's an usage to just spawn a child and does not wait for it. 
Do you think it is useful?

Reply:
for now, for what we want to achieve, it isn't useful. but I don't think it hurts having this additional feature.